### PR TITLE
Communication - Extract the Identity Client out of the Administration library

### PR DESCRIPTION
**Pull request by RezaJooyandeh**

### CODE REVIEW

File: sdk/communication/Azure.Communication.Administration/api/Azure.Communication.Administration.netstandard2.0.cs [hunk 1]
```diff
@@ -1,59 +1,10 @@
...
-        internal CommunicationUserToken() { }
-        public System.DateTimeOffset ExpiresOn { get { throw null; } }
-        public string Token { get { throw null; } }
-    }
     public partial class PhoneNumberAdministrationClient
     {
         protected PhoneNumberAdministrationClient() { }
         public PhoneNumberAdministrationClient(string connectionString, Azure.Communication.Administration.PhoneNumberAdministrationClientOptions? options = null) { }
-        public PhoneNumberAdministrationClient(System.Uri endpoint, Azure.AzureKeyCredential keyCredential, Azure.Communication.Administration.CommunicationIdentityClientOptions? options = null) { }
+        public PhoneNumberAdministrationClient(System.Uri endpoint, Azure.AzureKeyCredential keyCredential, Azure.Communication.Administration.PhoneNumberAdministrationClientOptions? options = null) { }
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/api/Azure.Communication.Administration.netstandard2.0.cs:7]: Good catch!

Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Administration/api/Azure.Communication.Administration.netstandard2.0.cs:7]: Right?! Compiler caught it 😋. One more reason why this was a good idea

File: sdk/communication/Azure.Communication.Identity/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+```C# Snippet:CreateCommunicationIdentityFromToken
+var endpoint = new Uri("https://my-resource.communication.azure.com");
+TokenCredential tokenCredential = new DefaultAzureCredential();
+var client = new CommunicationIdentityClient(endpoint, tokenCredential);
+```
+
+### Key concepts
+
+`CommunicationIdentityClient` provides the functionalities to manage user access tokens: creating new ones, renewing and revoking them.
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/README.md:54]: How can tokens be "renewed"? We can only create a replacement token, right?

File: sdk/communication/Azure.Communication.Identity/samples/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+languages:
+- csharp
+products:
+- azure
+- azure-communication
+name: Azure Communication Identity samples for .NET
+description: Samples for the Azure.Communication.Identity client library
+---
+
+# Azure Communication AdministIdentityration SDK samples
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/samples/README.md:12]:
```suggestion
# Azure Communication Identity SDK samples
```

File: sdk/communication/Azure.Communication.Identity/samples/Sample1_CommunicationIdentityClient.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+Response<CommunicationUserIdentifier> userResponse = client.CreateUser();
+CommunicationUserIdentifier user = userResponse.Value;
+Console.WriteLine($"User id: {user.Id}");
+```
+
+You will need to store the `identity` that is returned by Azure Communication Services. This is necessary for creating tokens or refreshing them in the future and to map your user identities with Azure Communication Services identities. The `identity` value should be treated as a secret.
+
+## Generate a user token
+
+ <!---: TODO: Update the website address for explaining the scopes -->
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/samples/Sample1_CommunicationIdentityClient.md:43]: We should fix those Todos soon

File: sdk/communication/Azure.Communication.Identity/src/Azure.Communication.Identity.csproj [hunk 5]
```diff
@@ -0,0 +1,43 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Description>
+      This client library enables working with the Microsoft Azure Communication Identity service.
+      For this release, see notes - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/communication/Azure.Communication.Identity/README.md and https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/communication/Azure.Communication.Identity/CHANGELOG.md.
+      Microsoft Azure Communication Identity quickstart - https://docs.microsoft.com/azure/communication-services/quickstarts/access-tokens?pivots=programming-language-csharp
+    </Description>
+    <AssemblyTitle>Azure Communication Identity Service</AssemblyTitle>
+    <Version>1.0.0-beta.4</Version>
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/src/Azure.Communication.Identity.csproj:9]: same question about version of first release

File: sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClientOptions.cs [hunk 6]
```diff
@@ -22,24 +21,13 @@ public class CommunicationIdentityClientOptions : ClientOptions
...
-            if (transport != default)
-                Transport = transport;
-
-            if (retryOptions != null)
-            {
-                Retry.Mode = retryOptions.Mode;
-                Retry.MaxRetries = retryOptions.MaxRetries;
-                Retry.Delay = retryOptions.Delay;
-                Retry.MaxDelay = retryOptions.MaxDelay;
-            }
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClientOptions.cs:42]: Why do we now longer support customizing Transport and Retry?

File: sdk/communication/Azure.Communication.Identity/src/autorest.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Azure.Communication.Administration
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/src/autorest.md:1]:
```suggestion
# Azure.Communication.Identity
```

File: sdk/communication/Azure.Communication.Identity/src/autorest.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+Run `dotnet msbuild /t:GenerateCode` to generate code.
+
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+``` yaml
+input-file:
+    -  https://raw.githubusercontent.com/Azure/azure-rest-api-specs/084de2711f77d12d644c7628b61cdd7634341ee8/specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/stable/2021-03-07/CommunicationIdentity.json
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/src/autorest.md:10]: Had we merged the new identity API version to master in C#?

So do we want to release with the new service version?

**Review by DominikMe — COMMENTED** [review]
Great! Found some nits

File: sdk/communication/Azure.Communication.Identity/CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# Release History
+
+## 1.0.0-beta.4 (Unreleased)
```
[the remark this answers is not in the document]
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Identity/CHANGELOG.md:3]: let's do 4

File: sdk/communication/Azure.Communication.Identity/src/Azure.Communication.Identity.csproj (hunk 5 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Identity/src/Azure.Communication.Identity.csproj:9]: Let's keep them in sync

File: sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClientOptions.cs (hunk 6 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClientOptions.cs:42]: We do. ClientOptions has public setters. The code that I deleted is very ancient. There is no reason to add another mechanism for setting them. It came up in the review as well.

**Pull request merged**